### PR TITLE
fix: do not treat subpackage paths as part of the current package

### PR DESCRIPTION
**Pull request by matt2e**
fixes https://github.com/TBD54566975/ftl/issues/1529

### CODE REVIEW

File: go-runtime/compile/schema_test.go [hunk 1]
```diff
@@ -398,46 +398,46 @@ func TestErrorReporting(t *testing.T) {
 	filename := filepath.Join(pwd, `testdata/failing/failing.go`)
 	actual := slices.Map(r.MustGet().Errors, func(e *schema.Error) string { return strings.TrimPrefix(e.Error(), filename+":") })
 	expected := []string{
-		`10:13-34: first argument to config and secret declarations must be the name as a string literal`,
```
Comment (matt2e) [inline go-runtime/compile/schema_test.go:401]: the new import has moved everything down a line...

File: go-runtime/compile/schema_test.go [hunk 2]
```diff
@@ -398,46 +398,48 @@ func TestErrorReporting(t *testing.T) {
...
+		`84:1-1: duplicate verb name "WrongResponse"`,
+		`90:2-12: struct field unexported must be exported by starting with an uppercase letter`,
+		`102:2-24: cannot attach enum value to BadValueEnum because it is a variant of type enum TypeEnum, not a value enum`,
+		`109:2-41: cannot attach enum value to BadValueEnumOrderDoesntMatter because it is a variant of type enum TypeEnum, not a value enum`,
+		`118:1-26: parent enum "ExportedTypeEnum" is exported, but directive "ftl:data" on "PrivateData" is not: all variants of exported enums that have a directive must be explicitly exported as well`,
+		`122:21-60: config and secret names must be valid identifiers`,
+		`128:1-26: only one directive expected for type alias`,
+		`144:1-35: type can not be a variant of more than 1 type enums (TypeEnum1, TypeEnum2)`,
+		`152:2-2: unsupported external type ftl/failing/child.ChildStruct`,
+		`152:2-6: unsupported type "ftl/failing/child.ChildStruct" for field "Data"`,
```
Comment (matt2e) [inline go-runtime/compile/schema_test.go:442]: the 2 errors for line 152 are what are actually being surfaced with the new test change

File: go-runtime/compile/schema_test.go [hunk 3]
```diff
@@ -398,46 +398,52 @@ func TestErrorReporting(t *testing.T) {
...
+		`118:1-26: parent enum "ExportedTypeEnum" is exported, but directive "ftl:data" on "PrivateData" is not: all variants of exported enums that have a directive must be explicitly exported as well`,
+		`122:21-60: config and secret names must be valid identifiers`,
+		`128:1-26: only one directive expected for type alias`,
+		`144:1-35: type can not be a variant of more than 1 type enums (TypeEnum1, TypeEnum2)`,
+		`152:2-2: unsupported type ftl/failing/child.ChildStruct from subpackage`,
+		`152:2-6: unsupported type "ftl/failing/child.ChildStruct" for field "Data"`,
+		`153:2-2: unsupported type ftl/failing/child.ChildString from subpackage`,
+		`153:2-8: unsupported type "ftl/failing/child.ChildString" for field "String"`,
+		`157:1-1: unsupported type ftl/failing/child.ChildStruct from subpackage`,
+		`157:1-33: unsupported type "struct{Name string}" for type alias`,
```
Comment (matt2e) [inline go-runtime/compile/schema_test.go:446]: errors for line 152 onwards are what are actually being surfaced with the new test change

**Comment by matt2e** [comment]
This wasn't as neat as I'd hoped. Originally I was hoping it'd just be a change to `isPathInPkg(path)`, but ended up needing additional checks each time. Otherwise it was unable to tell external refs (allowed) from subpackage refs (not allowed) because subpackages will naturally also start with `ftl/`

File: go-runtime/compile/schema.go [hunk 4]
```diff
@@ -1483,11 +1496,29 @@ func (p *parseContext) pathEnclosingInterval(start, end token.Pos) (pkg *package
 	return nil, nil, false
 }
 
+// isPathInPkg checks that the path is within the current package
+//
+// if it is in a subpackage, it will return false
 func (p *parseContext) isPathInPkg(path string) bool {
-	if path == p.pkg.PkgPath {
-		return true
+	if strings.Contains(path, "namedext") {
```
Comment (alecthomas) [inline go-runtime/compile/schema.go:1503]: Debugging?

**Review by alecthomas — APPROVED** [review]

File: go-runtime/compile/schema.go (hunk 4 above)
Comment (matt2e) [reply go-runtime/compile/schema.go:1503]: 🤦‍♂️

**Pull request merged**